### ISSUE TITLE
Intellij error while analyzing the Angular repository.

### ISSUE THREAD

**Issue by alorenzen** [dart-lang/sdk#32599]
Analyzer Feedback from IntelliJ

# Version information

- `IDEA IC-173.4301.25`
- `2.0.0-dev.39.0`
- `IC-173.4301.25, JRE 1.8.0_152-release-1024-b11x64 JetBrains s.r.o, OS Mac OS X(x86_64) v10.13.3 unknown, screens 3840x2160`

# Exception

Please append the contents of:
file:///private/var/folders/ls/1jj80hm96sb7j_wjh2tb2bnm005b3f/T/report.txt

```
Dart analysis server, SDK version 2.0.0-dev.39.0, server version 1.20.0, error: Analysis failed: /Users/alorenzen/angular/examples/github_issues/test/issue_body_test.dart
RangeError (byteOffset): Invalid value: Not in range 0..1860, inclusive: 2952790061
#0      _ByteDataView.getUint32 (dart:typed_data-patch/dart:typed_data/typed_data_patch.dart:4278)
#1      BufferContext._getUint32 (package:front_end/src/base/flat_buffers.dart:72)
#2      StringReader.read (package:front_end/src/base/flat_buffers.dart:663)
#3      _FbGenericList.[] (package:front_end/src/base/flat_buffers.dart:826)
#4      Object

**Comment by alorenzen** [comment]
Tried again this morning, and got a similar crash, but on a different file:
```
Dart analysis server, SDK version 2.0.0-dev.39.0, server version 1.20.0, error: Analysis failed: /Users/alorenzen/angular/_tests/test/core/linker/deferred_view_with_ngmodel.dart
RangeError (byteOffset): Invalid value: Not in range 0..1860, inclusive: 2952790061
#0      _ByteDataView.getUint32 (dart:typed_data-patch/dart:typed_data/typed_data_patch.dart:4278)
#1      BufferContext._getUint32 (package:front_end/src/base/flat_buffers.dart:72)
#2      StringReader.read (package:front_end/src/base/flat_buffers.dart:663)
#3      _FbGenericList.[] (package:front_end/src/base/flat_buffers.dart:826)
#4      Object

**Comment by bwilkerson** [comment]
@scheglov (in case you have any ideas)

**Comment by scheglov** [comment]
Looks like a damaged cache.
Remove `~/.dartServer/.analysis-driver` and restart Analysis Server.

**Comment by alorenzen** [comment]
I did that a few times, once just restarting Analysis Server, the other restarting Intellij completley, and am still getting these errors.

```
Dart analysis server, SDK version 2.0.0-dev.39.0, server version 1.20.0, error: Analysis failed: /Users/alorenzen/angular/dev/bin/pub.dart
RangeError (byteOffset): Invalid value: Not in range 0..1980, inclusive: 262186
#0      _ByteDataView.getUint32 (dart:typed_data-patch/dart:typed_data/typed_data_patch.dart:4278)
#1      BufferContext._getUint32 (package:front_end/src/base/flat_buffers.dart:72)
#2      StringReader.read (package:front_end/src/base/flat_buffers.dart:663)
#3      _FbGenericList.[] (package:front_end/src/base/flat_buffers.dart:826)
#4      Object

**Comment by scheglov** [comment]
What version of Angular plugin do you use?
@MichaelRFairhurst is this version with the cache race condition fixed?

**Comment by MichaelRFairhurst** [comment]
This isn't a damaged cache...it's fixed but unpublished still... I'll be
publishing this week hopefully.

The problem is from latest intellij using the navigation API in a way that
crashes

**Comment by bwilkerson** [comment]
@jwren @alexander-doroshko

**Comment by MichaelRFairhurst** [comment]
I should be clearer that the problem was on my end and not the intelliJ
end. But it broke without me noticing at some point since I hadn't tested
it on the EAP.

**Comment by MichaelRFairhurst** [comment]
Fixed in 0.0.14
